akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala

```diff
@@ -287,15 +287,9 @@ private[akka] class RemoteWatcher(
    * does not exist.
    */
   def reWatch(address: Address): Unit =
```

Comment:
Is this still necessary? My intuition is that we should now have all the information in one place to just fire DeathWatchNotification upon system UID change, but I might be missing something. @patriknw ?